### ISSUE TITLE
Format of generic throwing closure type breaks compile

### ISSUE THREAD

**Issue by CrownedPhoenix** [nicklockwood/SwiftFormat#1899]
```swift 

struct Thing<T> {
    init(_: T) {}
}

let a = Thing<[(Int) throws -> [Int]]>([])
let b = Thing<((Int) throws -> [Int])?>(nil)
```
is formatted to
```swift
struct Thing<T> {
    init(_: T) {}
}

let a = Thing < [(Int) throws -> [Int]] > []
let b = Thing < ((Int) throws -> [Int])? > nil
```
which does not compile

It appears that the `throws` keyword is the catalyst for this issue.
Removing `throws` from the closures prevent the failed format.

**Comment by CrownedPhoenix** [comment]
Sorry I need to edit this I just realized it does compile. I reduced the example too much.

**Comment by nicklockwood** [comment]
@CrownedPhoenix fixed in 0.54.6
